packages/kit/src/exports/vite/index.js

```diff
@@ -348,6 +348,22 @@ async function kit({ svelte_config }) {
 		 * Stores the final config.
 		 */
 		configResolved(config) {
+			// we search for this plugin by name because we can't detect it
+			// since it doesn't directly modify the https config unlike the mkcert plugin
+			const vite_basic_ssl = config.plugins.find(({ name }) => name === 'vite:basic-ssl');
+
+			// by default, when enabling HTTPS in Vite, it also enables HTTP/2
+			// however, undici's Headers implementation does not like the HTTP/2 headers
+			// we set a no-op proxy config to force Vite to downgrade to TLS-only
```

Comment:
I wonder if this logic would better live in Vite. It seems like every framework would encounter this